charts/osm/templates/cleanup-hook.yaml

```diff
@@ -0,0 +1,28 @@
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ .Release.Name }}-cleanup
+  namespace: {{ include "osm.namespace" . }}
+  labels:
+    {{- include "osm.labels" . | nindent 4 }}
+  annotations:
+    helm.sh/hook: post-delete
+    helm.sh/hook-delete-policy: before-hook-creation,hook-succeeded
+spec:
+  template:
+    metadata:
+      name: {{ .Release.Name }}-cleanup
+      labels:
+        {{- include "osm.labels" . | nindent 8 }}
+    spec:
+      serviceAccountName: {{ .Release.Name }}-cleanup
+      restartPolicy: Never
+      containers:
+        - name: garbage-collector
+          image: bitnami/kubectl
+          imagePullPolicy: IfNotPresent
+          command:
+            - sh
+            - -c
+            - >
+             kubectl delete --ignore-not-found configmap -n '{{ include "osm.namespace" . }}' osm-config;
```

Comment:
@nojnhuh this is already great!

I wonder if we could expand the scope of the delete operation and point it to all remaining OSM resources. We can leverage the changes in #3068 for this.

Something like
```bash
kubectl delete all -n $NS --selector "app.kubernetes.io/name=openservicemesh"
```

Reply:
@draychev I mentioned why I'm not sure that's the best approach (for now) here: https://github.com/openservicemesh/osm/issues/2931#issuecomment-812052480. 

tl;dr Helm already does most of the work, it's not as simple as a one-liner, and it requires much more extensive RBAC permissions.

Do you think the benefits to the all-in-one approach currently outweigh those concerns?

Reply:
@draychev I'm going to merge this for now. If we need something more generic we can readdress this then.